src/NServiceBus.RabbitMQ.Tests/MessageConverterTests.cs

```diff
@@ -147,36 +146,71 @@ public void TestCanHandleStringHeader()
                     MessageId = "Blah",
                     Headers = new Dictionary<string, object>
                     {
-                        {"Foo", "ni"}
+                        {"short", 42},
+                        {"int", 42},
+                        {"long", 42},
+                        {"decimal", 42m},
+                        {"sbyte", (sbyte)42},
+                        {"double", 42d},
+                        {"single", 42f},
+                        {"bool", true }
                     }
                 }
             };
 
             var headers = converter.RetrieveHeaders(message);
 
             Assert.NotNull(headers);
-            Assert.AreEqual("ni", headers["Foo"]);
+            Assert.AreEqual("42", headers["short"]);
+            Assert.AreEqual("42", headers["int"]);
+            Assert.AreEqual("42", headers["long"]);
+            Assert.AreEqual("42", headers["decimal"]);
+            Assert.AreEqual("42", headers["sbyte"]);
+            Assert.AreEqual("42", headers["double"]);
+            Assert.AreEqual("42", headers["single"]);
+            Assert.AreEqual("True", headers["bool"]);
         }
 
         [Test]
-        public void TestCanHandleStringArrayListsHeader()
```

Comment:
Same reply as to why I removed the `string` path from the method. We will never get an ArrayList passed to us from the Rabbit client.